price-estimator/src/main.rs

```diff
@@ -0,0 +1,96 @@
+use core::{
+    contracts::{stablex_contract::StableXContractImpl, web3_provider},
+    http::HttpFactory,
+    metrics::{HttpMetrics, MetricsServer},
+    util::FutureWaitExt as _,
+};
+use ethcontract::PrivateKey;
+use prometheus::Registry;
+use std::{num::ParseIntError, path::PathBuf, sync::Arc, thread, time::Duration};
+use structopt::StructOpt;
+use tokio::runtime;
+use url::Url;
+use warp::Filter;
+
+#[derive(Debug, StructOpt)]
+#[structopt(name = "price estimator", rename_all = "kebab")]
+struct Options {
+    /// The log filter to use.
+    ///
+    /// This follows the envlogger syntax (e.g. 'info,driver=debug').
+    #[structopt(
+        long,
+        env = "LOG_FILTER",
+        default_value = "warn,driver=info,price_estimator=info"
+    )]
+    log_filter: String,
+
+    /// The Ethereum node URL to connect to. Make sure that the node allows for
+    /// queries without a gas limit to be able to fetch the orderbook.
+    #[structopt(short, long, env = "NODE_URL")]
+    node_url: Url,
+
+    /// The timeout in seconds of web3 JSON RPC calls.
+    #[structopt(
+        long,
+        env = "TIMEOUT",
+        default_value = "10",
+        parse(try_from_str = duration_secs),
+    )]
+    timeout: Duration,
+
+    #[structopt(long, env = "ORDERBOOK_FILE", parse(from_os_str))]
+    orderbook_file: Option<PathBuf>,
+
+    #[structopt(
+        long,
+        env = "ORDERBOOK_UPDATE_INTERVAL",
+        default_value = "100",
+        parse(try_from_str = duration_secs),
+    )]
+    orderbook_update_interval: Duration,
+}
+
+fn main() {
+    let options = Options::from_args();
+    env_logger::init();
+    log::info!(
+        "Starting price estimator with runtime options: {:#?}",
+        options
+    );
+
+    let driver_http_metrics = setup_driver_metrics();
```

Comment:
Is this really "driver" metrics or is this just `http_metrics`? Looks like we may have recycled a term from the driver code here.

Reply:
I called it driver because presumably we will also have price estimator metrics but I wouldn't worry about the name yet.